template/ios/HelloWorld/AppDelegate.swift

```diff
@@ -0,0 +1,28 @@
+import UIKit
+import React
+import React_RCTAppDelegate
+
+@main
+class AppDelegate: RCTAppDelegate {
+  override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
+    self.moduleName = "HelloWorld";
+    
+    // You can add your custom initial props in the dictionary below.
+    // They will be passed down to the ViewController used by React Native.
+    self.initialProps = [:];
+    
+    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
+  }
+  
+  override func sourceURL(for bridge: RCTBridge) -> URL? {
+    self.bundleURL()
+  }
+  
```

Comment:
will override the `newArchEnabled` flag works with this setup?

That flag is not part of the header file for RCTAppDelegate, and in objective-C it works thanks to dynamic dispatch.
I don't think that it will work in Swift: I think that the swift compiler will complain that you can't `override` a function that is not declared in the header file.

If that's the case, we first need to update the `RCTAppDelegate.h` file.

Reply:
Thanks for your review.

So the `newArchEnabled` is not part of the public API should users use the `newArchEnabled` flag or the publically available: 
 
![CleanShot 2024-08-22 at 18 28 22@2x](https://github.com/user-attachments/assets/b0943e83-5a1f-4f8e-87b6-cb659685bce5)

We can expose the `newArchEnabled` flag too

Reply:
I think that's the best. The plan would be to remove these three flags, at some point, as the New Architecture is meant to work as a whole, not to disable pieces of it.

Reply:
This just got released in the first RC of 0.76, I think we can get back to this PR after releasing stable version of 0.76 wdyt?